lib/providers/user_provider.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:logger/logger.dart';
+import 'package:regatta_buddy/models/user_data.dart';
+import 'package:regatta_buddy/services/authentication_service.dart';
+import 'package:regatta_buddy/utils/logging/logger_helper.dart';
+
+class UserProvider with ChangeNotifier {
+  Logger logger = getLogger('UserProvider');
+  final AuthenticationService loginService = Get.find<AuthenticationService>();
+
+  UserData? _user;
+
+  UserData? get user {
+    initializeUserDataIfAuthenticated();
+    return _user;
+  }
+
+  void setUser(UserData user) {
+    _user = user;
+    notifyListeners();
+  }
+
+  void initializeUserDataIfAuthenticated() {
+    if (_user != null) {
+      return;
+    }
+
+    loginService.fetchCurrentUserData().then((userData) {
```

Comment:
moze byc ze userData jest nullem jak user nie jest?

Reply:
ta,  /// Returns [UserData] if successful, otherwise returns null. ale zauwazylem ze tu zostala dawna nazwa serwisu - loginService. W pozniejszym etapie zamienilem na AuthenticationService